server/src/main/java/org/loomdev/loom/entity/player/PlayerImpl.java

```diff
@@ -174,13 +176,13 @@ public int getProtocolVersion() {
     @Override
     @NotNull
     public Optional<Component> getTabListName() {
-        return Optional.ofNullable(TextTransformer.toLoom(getMinecraftEntity().tabListName));
+        return Optional.ofNullable(tabListName);
     }
 
     @Override
     public void setTabListName(@NotNull Component name) {
-        getMinecraftEntity().tabListName = TextTransformer.toMinecraft(name);
-        getMinecraftEntity().updateTabListName();
+        tabListName = name;
+        updateTabListName();
```

Comment:
Why not just throw the code for updateTabListName() into this method instead of delegating it to a separate method (since it's only called once here)?